[MHV-45649] Updated Page title tags across SM

vsaleem opened the pull request:
## Summary

Issue 1) Page Title Tags
- Updated Page Title Tags across Secure Messaging so that they are descriptive and unique.
- Use navigation or click on links to view dynamic updates to the page titles
- Most of the updates were handled on the component (top-level) so that the page titles are consistent with the user's current location

Issue 2) Updated Heading Hierarchy
- Heading tags aren't properly nested on 'Start a new message' page - (https://github.com/department-of-veterans-affairs/va.gov-team/issues/59496)
- Heading tags aren't properly nested on 'Interstitial' page - (https://github.com/department-of-veterans-affairs/va.gov-team/issues/59497)

- Va.gov - Secure Messaging


## Related issue(s)

- www.jira.devops.va.gov/browse/MHV-45649
- www.jira.devops.va.gov/browse/MHV-46119

## Testing done

- Prior to change, page title tags for sm were not unique and produced a standard tag for all pages
- Use navigation or click on links to view dynamic updates to the page titles
- Unit tests / SQA testing


## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |![image](https://github.com/department-of-veterans-affairs/vets-website/assets/59583325/ce92b9f3-ea26-4052-b63c-c1008a08abb6)|
| Desktop |        |![image](https://github.com/department-of-veterans-affairs/vets-website/assets/59583325/49d9173a-c465-43d0-b65f-1f8bf5e595e0)|

## What areas of the site does it impact?

*Va.gov - Secure Messaging*

## Acceptance criteria

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

Commit:
[MHV-45649] Updated page title tags via navigation

- added effect to dynamically update page title tags from top-level in navigation

Commit:
[MHV-45649] Updated nested page title tags across SM

- Updated nested page title tags
- Added updatePageTitle and convertPathNameToTitleCase helpers

Commit:
[MHV-45649] Updated unit tests; cleanup

oleksii-morgun-va commented on src/applications/mhv/secure-messaging/components/FoldersList.jsx:
```diff
@@ -1,12 +1,16 @@
+  const handleFolderPageTitles = newTitle => {
+    updatePageTitle(`${newTitle} ${PageTitles.PAGE_TITLE_TAG}`);
+  };
+
```
I am curious if this is necessary if you are already handling this in `Folders.jsx`

oleksii-morgun-va commented on src/applications/mhv/secure-messaging/components/MessageDetailBlock.jsx:
```diff
@@ -61,15 +61,18 @@ const MessageDetailBlock = props => {
       focusElement(document.querySelector('h1'));
+      document.title = `${categoryLabel}: ${subject} ${
+        PageTitles.PAGE_TITLE_TAG
+      }`;
```
I am wondering if it would make sense to handle all titles for Thread Details view in `ThreadDetails.jsx` component, especially since you are already handling some scenarios there?

oleksii-morgun-va commented on src/applications/mhv/secure-messaging/components/MessageList/FolderHeader.jsx:
```diff
@@ -38,6 +40,16 @@ const FolderHeader = props => {
+      }
+    },
+    [folder],
+  );
```
I am guessing it is working fine as is, but perhaps would make a better sense to incorporate on a parent level like `FolderThreadListView.jsx` as it appears to be more relevant to entire container rather than just his header component. Just a suggestion, not necessarily it is the best way to go. Leaving up to you to make that call.

oleksii-morgun-va commented on src/applications/mhv/secure-messaging/components/Navigation.jsx:
```diff
@@ -18,14 +20,40 @@ const Navigation = () => {
+      // initial page title
+      updatePageTitle(pageTitle());
+    },
+    [defaultInitialPageTitle, initialPageTitle, location.pathname],
```
Same as a comment for `FolderHeader`. Sent can be handled in `FolderThreadListView` and Start a new message is more appropriate to a `Compose` container, what do you think?

oleksii-morgun-va commented on src/applications/mhv/secure-messaging/tests/containers/Folders.unit.spec.jsx:
```diff
@@ -13,12 +15,28 @@ describe('Folders Landing Page', () => {
+      );
+    });
+  });
+
```
Do you think it is feasible to expand test coverage to other pages beyond a landing page and My Folders?

Commit:
[MHV-46119] SHOULD for Phase 0 updates

- Updated Compose page heading level structure to h2

- Removed h2 role within main tag from InterstitalPage (caused confusion and not exclusively necessary.)

Commit:
[MHV-45649] Revised PageTitles handling

- Switched page title tag handling to be on component parent level instead of navigation level

- Updated unit test

Commit:
[MHV-45649] FolderThreadListView unit test

- tests for page title tags on FolderThreadListView component

Commit:
Merge branch 'main' into 45649-Page-Title-Tags-across-SM

Commit:
[MHV-45649] Fixed Custom Folder Page Titles

- added useEffect for custom folder titles

-cleanup

Commit:
Merge branch 'main' into 45649-Page-Title-Tags-across-SM

Commit:
Merge branch 'main' into 45649-Page-Title-Tags-across-SM

Commit:
[MHV-45649] Revised FolderHeader unit tests

- updated some unit tests that were failing; Revised FolderHeader unit tests because previous test no longer applied to the current functionality

- cleanup; added constants

Commit:
[MHV-45649] Cleanup unit test

Commit:
Merge branch 'main' into 45649-Page-Title-Tags-across-SM

Commit:
Merge branch 'main' into 45649-Page-Title-Tags-across-SM

Commit:
[MHV-45649] Using updatePageTitle helper for doc.title
